examples/code/role_playing_multiprocess.py

```diff
@@ -134,7 +129,7 @@ def generate_data(language_idx: int, language_name: str, domain_idx: int,
                 f"{user_response.info['termination_reasons'][0]}")
             break
 
-        user_agent.update_messages(user_response.msg)
+        user_agent.submit_critic_choice(user_response.msg)
```

Comment:
Update_messages is too unspecific. First off, "update" means any change to the messages. Secondly, the historic messages are a foreign concept to a user of the agent api. It could be a private _append_to_messages(). But still the original name implies that a user can go wild with its use. I would restrict the api to several meaningful uses like "submit_critic_response" and if you want "force_append_message" which has to be marked as not recommended.